### PR TITLE
*: add OpenShift Dockerfiles

### PR DESCRIPTION
**Pull request by hexfusion**
Signed-off-by: Sam Batschelet <sbatsche@redhat.com>

### CODE REVIEW

File: Dockerfile.kube-client-agent.openshift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.10 AS builder
+
+COPY . /go/src/github.com/coreos/kubecsr
```
Comment (ericavonb) [inline Dockerfile.kube-client-agent.openshift]: 's/coreos/openshift/' 🙂

File: Dockerfile.kube-client-agent.openshift [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN cd /go/src/github.com/coreos/kubecsr \
```
Comment (ericavonb) [inline Dockerfile.kube-client-agent.openshift]: optional: you can simplify it by setting the workdir
```Dockerfile
WORKDIR /go/src/github.com/openshift/kubecsr
COPY . .
RUN make bin/kube-client-agent
```

File: Dockerfile.kube-client-agent.openshift [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  && make bin/kube-client-agent
+
+# stage 2
+FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base
+
+ENTRYPOINT ["/usr/bin/kube-client-agent"]
+
+COPY --from=builder /go/src/github.com/coreos/kubecsr/bin/kube-client-agent /usr/bin/
```
Comment (ericavonb) [inline Dockerfile.kube-client-agent.openshift]: 's/coreos/openshift/'

File: Dockerfile.kube-etcd-signer-server.openshift [hunk 4]
```diff
@@ -0,0 +1,17 @@
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.10 AS builder
+
+COPY . /go/src/github.com/coreos/kubecsr
```
Comment (ericavonb) [inline Dockerfile.kube-etcd-signer-server.openshift]: s/coreos/openshift/

File: Dockerfile.kube-etcd-signer-server.openshift [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN cd /go/src/github.com/coreos/kubecsr \
```
Comment (ericavonb) [inline Dockerfile.kube-etcd-signer-server.openshift]: same comment re: workdir

File: Dockerfile.kube-client-agent.openshift [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+LABEL io.k8s.display-name="kube-client-agent" \
+      io.k8s.description="This client agent generates a valid CSR according to the configuration flags provided to it." \
+      maintainer="Sam Batschelet <sbatsche@redhat.com>"
```
Comment (ericavonb) [inline Dockerfile.kube-client-agent.openshift]: Do you need these too? for getting the images into the release
```Dockerfile
io.openshift.tags="openshift" \
io.openshift.release.operator=true \
```

File: Dockerfile.kube-etcd-signer-server.openshift [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+# stage 2
+FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base
+
+ENTRYPOINT ["/usr/bin/kube-etcd-signer-server"]
+
+COPY --from=builder /go/src/github.com/coreos/kubecsr/bin/kube-etcd-signer-server /usr/bin/
+
+LABEL io.k8s.display-name="kube-etcd-signer-server" \
+      io.k8s.description="The certificate signer approves a given valid CSR provided by the client agent and returns a signed certificate." \
+      maintainer="Sam Batschelet <sbatsche@redhat.com>"
```
Comment (ericavonb) [inline Dockerfile.kube-etcd-signer-server.openshift]: same comment re `io.openshift.tags="openshift"` and `io.openshift.release.operator=true` labels

File: Dockerfile.kube-client-agent.openshift (hunk 6 above)
Comment (hexfusion) [reply Dockerfile.kube-client-agent.openshift]: yes that is my understanding

File: Dockerfile.kube-etcd-signer-server.openshift (hunk 7 above)
Comment (hexfusion) [reply Dockerfile.kube-etcd-signer-server.openshift]: see above

File: Dockerfile.kube-client-agent.openshift (hunk 1 above)
Comment (hexfusion) [reply Dockerfile.kube-client-agent.openshift]: good catch!

**Comment by hexfusion** [comment]
@ericavonb we import coreos org so I think we need to keep it :( unless we want to make a bunch of changes.

```
Step 4/8 : RUN make bin/kube-client-agent
 ---> Running in 37179a78621e
cmd/kube-client-agent/request.go:9:2: cannot find package "github.com/coreos/kubecsr/pkg/certagent" in any of:
	/go/src/github.com/openshift/kubecsr/vendor/github.com/coreos/kubecsr/pkg/certagent (vendor tree)
	/usr/local/go/src/github.com/coreos/kubecsr/pkg/certagent (from $GOROOT)
	/go/src/github.com/coreos/kubecsr/pkg/certagent (from $GOPATH)
make: *** [bin/kube-client-agent] Error 1
```

File: Dockerfile.kube-client-agent.openshift (hunk 2 above)
Comment (hexfusion) [reply Dockerfile.kube-client-agent.openshift]: good advice thanks!

**Comment by ericavonb** [comment]
@hexfusion I think this should do it: https://github.com/openshift/kubecsr/pull/3

**Comment by hexfusion** [comment]
@ericavonb :)! The only issue with that is we need to rebase each time we sync up with upstream. My understanding is that we are not really changing the upstream or maintaining a fork persay we are just mirroring upstream and repackaging it. Same way RHEL would release etcd for example. I think the least amount of changes we make the better in that regard. Because tomorrow I will have forgot!

**Comment by hexfusion** [comment]
Going to merge this so release has something to chew on we can revisit #3, thanks for the review.

**Pull request merged**